docs/provisioning/external-silos.md

```diff
@@ -0,0 +1,154 @@
+# Advanced - Provisioning a setup with _external_ silos
+
+
+## Contents
+Many real-world Federated Learning (FL) applications will rely on silos that are not in the same Azure tenant as the orchestrator. This is the case for example when the silos are owned by different companies. Furthermore, those silos might not even be in Azure at all - they might be on different cloud platforms, or on-premises.
+
+We refer to those types of silos as _external_ silos. The goal of this document is to **provide guidance on how to provision a FL setup with _external_ silos.**
+
+
+## Nomenclature
+- `FL Admin`: this is the person who is responsible for provisioning the FL setup and ensuring the security. He or she works for Contoso, the company who wants to train a model using FL.
+- `Silo Admin`: this is the person who owns the compute and the data of a given silo. He or she works for Fabrikam, a company that has agreed to share their data with Contoso for FL purposes. 
+- `Subscription`: in all that follows, when we talk about an _Azure subscription_ we mean the subscription where the Azure ML workspace and the orchestrator will be deployed. This subscription belongs to Contoso.
+- For common FL terms such as `silo` or `orchestrator`, please refer to the [glossary](../glossary.md).
+
+
+## Prerequisites
+- **Some Kubernetes (k8s) cluster** (at least one) with version <= 1.24.6, either on-premises, or in Azure (in a different tenant from that of the orchestrator). The cluster should have a minimum of 4 vCPU cores and 8-GB memory
+  - For creating a k8s cluster on-premises one can use [Kind](https://kind.sigs.k8s.io/), for instance.
+  - For creating a k8s cluster in Azure (in a different tenant otherwise we'd be dealing with _internal_ silos), one can use [Azure Kubernetes Service (AKS)](https://portal.azure.com/#create/microsoft.aks).
+- FL Admin needs to have **Owner** role in the subscription, or at least in the resource group where the Azure ML workspace will be created. This is because some Role Assignments will have to be created.
+- Silo Admin must be given (temporary) Contributor role to the subscription, or at least the [Azure Arc Onboarding](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#kubernetes-cluster---azure-arc-onboarding) built-in role. This is because one step will require access to both the orchestrator subscription, and to the k8s cluster. It assumed that _the FL Admin shouldn't have direct access to the k8s cluster_.
+
+
+## Procedure
+
+### A. Create the Azure ML workspace and the orchestrator
```

Comment:
To clarify the procedure, I suggest creating sections entirely dedicated to one role only.

Example:

A. FL Admin creates the Azure ML workspace and orchestrator

B. FL Admin prepared the integration of external silo

C. Silo Admin connects...

D. FL Admin ....

This way, it avoids having to refer to each role too much, and it can also be used between FL Admin and Silo Admin to understand each of their roles and the "hand-off" between there steps.

Reply:
done